src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,670 @@
+  bool RX;                   // Radio reception active            (TRUE = reception)
+  bool TX;                   // Radio transmission active         (TRUE = transmission)
+  bool RX_active;            // Radio reception on active station (TRUE = reception)
+  bool RX_standy;            // Radio reception on passive        (standby) station
+  bool lowBAT;               // Battery low flag                  (TRUE = Batt low)
+  bool TXtimeout;            // Timeout while transmission (2Min)
+} Radio_t;
+
+IntConvertStruct CRC;
+IntConvertStruct sFrequency;
+IntConvertStruct sStatus;
+
+volatile bool bSending = false;
+//End Einige Konstanten und Ausdruecke, das AR62xx hat ein Binaerprotokoll
+
+
+/**
+ * AR62xx device class.
+ *
+ * This class provides the interface to communicate with the AR62xx radio.
+ * The driver retransmits messages in case of a failure.
+ */
+class AR62xxDevice final : public AbstractDevice {
+  static constexpr unsigned CMD_TIMEOUT = 250; //!< Command timeout in millis.
+  static constexpr unsigned NR_RETRIES = 3; //!< Number of tries to send a command.
+
+  static constexpr char STX = 0x02; //!< Command start character.
+  static constexpr char ACK = 0x06; //!< Command acknowledged character.
+  static constexpr char NAK = 0x15; //!< Command not acknowledged character.
+  static constexpr char NO_RSP = 0; //!< No response received yet.
+
+  static constexpr size_t MAX_NAME_LENGTH = 10; //!< Max. radio station name length.
+  //! Port the radio is connected to.
+  Port &port;
+  //! Expected length of the message just receiving.
+  //size_t expected_msg_length{};
+  //! Buffer which receives the messages send from the radio.
+  StaticFifoBuffer<uint8_t, 256u> rx_buf;
+  //! Last response received from the radio.
+  uint8_t response;
+  //! Condition to signal that a response was received from the radio.
+  Cond rx_cond;
+  //! Mutex to be locked to access response.
+  std::unique_lock<Mutex> response_mutex;
+  // RadioPara aus LK8000
+  Radio_t RadioPara;
+
+public:
+  /**
+   * Constructor of the radio device class.
+   *
+   * @param _port Port the radio is connected to.
+   */
+  AR62xxDevice(Port &_port);
+
+private:
+  /**
+   * Sends a message to the radio.
+   *
+   * @param msg Message to be send to the radio.
+   */
+  bool Send(const uint8_t *msg, unsigned msg_size, OperationEnvironment &env);
+
+  /*
+   * Called by "PutActiveFrequency"
+   * doing the real work
+   */
+  bool AR620xPutFreqActive(double Freq, const TCHAR *StationName, OperationEnvironment &env);
+
+  /*
+   * Called by "PutStandbyFrequency"
+   * doing the real work
+   */
+  bool AR620xPutFreqStandby(double Freq, const TCHAR *StationName, OperationEnvironment &env);
+
+  /*
+   * Creates the correct index for a given readable frequency
+   */
+  uint16_t Frq2Idx(double fFreq);
+
+  /*
+   * Creates a correct frequency-number given by the index
+   */
+  double Idx2Freq(uint16_t uFreqIdx);
```

Comment:
What is an "index"?

Reply:
renamed to id, added documentation about the id